stdlib/REPL/src/REPL.jl

```diff
@@ -1111,6 +1111,31 @@ function setup_interface(
                 edit_insert(s, '?')
             end
         end,
+        ']' => function (s::MIState,o...)
+            if isempty(s) || position(LineEdit.buffer(s)) == 0
+                pkgid = Base.PkgId(Base.UUID("44cfe95a-1eb2-52ea-b672-e2afdf69b78f"), "Pkg")
+                if Base.locate_package(pkgid) !== nothing # Only try load Pkg if we can find it
+                    Pkg = Base.require(pkgid)
+                    # Pkg should have loaded its REPL mode by now, let's find it so we can transition to it.
+                    pkg_mode = nothing
+                    for mode in repl.interface.modes
+                        if mode isa LineEdit.Prompt && mode.complete isa Pkg.REPLMode.PkgCompletionProvider
```

Comment:
Can we do this without encoding the name?
It looks like for everything else we don't need to know about Pkg in Base which is great since we don't need to declare a compat bound ;)

Maybe we define a reference that Pkg sets in Base? If that reference is not set we do the load dance.

Reply:
We used to load Pkg automatically as part of launching the default REPL, which was removed when Pkg was added to the system image. We could likely still do that approach in Base (like InteractiveUtils, and the REPL itself)?

Reply:
> It looks like for everything else we don't need to know about Pkg in Base 

This is not Base.

> Maybe we define a reference that Pkg sets in Base? If that reference is not set we do the load dance.

What's the advantage of that over the current way?



> We could likely still do that approach in Base (like InteractiveUtils, and the REPL itself)?

I don't really want to load a package manager just because I load a REPL, no?

Reply:
> What's the advantage of that over the current way?

That REPL doesn't need to have a written reference to an internal implementation detail of Pkg. You would be more robust if someone renamed the struct.

Reply:
So you want a function in Pkg

```
REPLMode_PkgCompletionProvider() = REPLMode.PkgCompletionProvider
```

and use

```
mode.complete isa Pkg.REPLMode_PkgCompletionProvider()
```

instead? I mean, I guess but it doesn't seem to be a very big difference in practice.